### PR TITLE
Fix database.useEmulator typing.

### PR DESCRIPTION
**Pull request by yuchenshi**
Adds correct typing for new argument introduced in #4792.

### CODE REVIEW

File: packages/firebase/index.d.ts [hunk 1]
```diff
@@ -7054,6 +7136,8 @@ declare namespace firebase.database {
     logger?: boolean | ((a: string) => any),
     persistent?: boolean
   ): any;
+
+  export type EmulatorMockTokenOptions = firebase.EmulatorMockTokenOptions;
```
Comment (Feiyang1) [inline packages/firebase/index.d.ts:7140]: I think we can just use `firebase.EmulatorMockTokenOptions` without needing the alias

Comment (yuchenshi) [reply packages/firebase/index.d.ts:7140]: We already released a version with `firebase.firestore.EmulatorMockTokenOptions` and I think we should probably match that for `database` for consistency. WDYT?

Comment (Feiyang1) [reply packages/firebase/index.d.ts:7140]: hmm, good point. Ok, let do that.

**Review by Feiyang1 — APPROVED** [review]

**Review by egilmorez — APPROVED** [review]

File: packages/firebase/index.d.ts [hunk 2]
```diff
@@ -5711,8 +5786,15 @@ declare namespace firebase.database {
      *
      * @param host the emulator host (ex: localhost)
      * @param port the emulator port (ex: 8080)
+     * @param options.mockUserToken - the mock auth token to use for unit testing Security Rules
```
Comment (hsubox76) [inline packages/firebase/index.d.ts:5789]: The jsdoc format read by the "old" docgen tool doesn't like dashes (`-`).
The dashes are required in the exp doc comments though. Sorry, confusing.

**Review by hsubox76 — APPROVED** [review]
Approved pending one thing (see comment)

**Pull request merged**